### PR TITLE
Update Macaw to 0.2.24 and set its run mode

### PR DESCRIPTION
**Pull request by crisptrutski**
Bumps Macaw to 0.2.23 and synchronises its run mode with Metabase.

This should have two consequences:

- Remove the large stacktrace from CI
- Type errors if NQA is passed data in the wrong shape

### CODE REVIEW

**Comment by crisptrutski** [comment]
The latest version of Macaw overrides the global Malli registry in Metabase, we'll need to rework this machinery.

**Comment by crisptrutski** [comment]
One idea is to move all the Malli stuff into a library

**Comment by crisptrutski** [comment]
Tying things up

File: src/metabase/query_analysis/native_query_analyzer/impl.cljc [hunk 1]
```diff
@@ -51,7 +51,7 @@
      :quotes-preserve-case? (not (contains? #{:mysql :sqlserver} driver))
      :features              {:postgres-syntax        (isa? driver/hierarchy driver :postgres)
                              :square-bracket-quotes  (= :sqlserver driver)
-                             :unsupported-statements true
+                             :unsupported-statements false
```
Comment (crisptrutski) [inline src/metabase/query_analysis/native_query_analyzer/impl.cljc:54]: There's a surprising regression without this, where basic functions like COUNT suddenly return unsupported ASTs.

I don't understand why, and I'm not stepping into this rabbit hole.

Failing closed is probably the right choice in retrospect anyway.

**Review by qnkhuat — APPROVED** [review]

**Pull request merged**